extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -857,12 +857,17 @@ export class ProjectsController {
 			return;
 		}
 
-		// TODO: swap this out and hookup to "Move" file/folder api
-		// rename the file
-		const newFilePath = path.join(path.dirname(file?.fsUri.fsPath!), `${newFileName}.sql`);
-		await fs.rename(file?.fsUri.fsPath!, newFilePath);
-		await project.exclude(file!);
-		await project.addExistingItem(newFilePath);
+		const newFilePath = path.join(path.dirname(utils.getPlatformSafeFileEntryPath(file?.relativePath!)), `${newFileName}.sql`);
+		const sqlProjectsService = await utils.getSqlProjectsService();
+
+		if ((await sqlProjectsService.getSqlObjectScripts(project.projectFilePath)).scripts.includes(file!.relativePath)) {
+			await sqlProjectsService.moveSqlObjectScript(project.projectFilePath, newFilePath, file!.relativePath)
+		} else if ((await sqlProjectsService.getPreDeploymentScripts(project.projectFilePath)).scripts.includes(file!.relativePath)) {
+			await sqlProjectsService.movePreDeploymentScript(project.projectFilePath, newFilePath, file!.relativePath)
+		} else if ((await sqlProjectsService.getPostDeploymentScripts(project.projectFilePath)).scripts.includes(file!.relativePath)) {
```

Comment:
As discussed in chat, I'll work on adding the file type info (script/pre/post deploy) in the project tree to avoid all the get calls, then a cleanup task later to simplify to a single Move() api instead of each collection having their own.